src/mongo/db/audit/audit.cpp

```diff
@@ -214,6 +227,101 @@ namespace audit {
         SimpleMutex _mutex;
     };    
 
+    // Writes audit events to a json file
+    class JSONAuditLog: public FileAuditLog {
+
+        class Adapter: public AuditLogFormatAdapter {
+            const std::string str;
+
+        public:
+            Adapter(const BSONObj &obj)
+                : str(mongoutils::str::stream() << obj.jsonString() << "\n") {}
+
+            virtual const char *data() const override {
+                return str.c_str();
+            }
+            virtual unsigned size() const override {
+                return str.size();
+            }
+        };
+
+    protected:
+        virtual AuditLogFormatAdapter *getAdapter(const BSONObj &obj) const override {
+            return new Adapter(obj);
+        }
+    public:
+        JSONAuditLog(const std::string &file, const BSONObj &filter)
+            : FileAuditLog(file, filter) {}
+    };
+
+    // Writes audit events to a bson file
+    class BSONAuditLog: public FileAuditLog {
+
+        class Adapter: public AuditLogFormatAdapter {
+            const BSONObj &obj;
```

Comment:
Yes, I think it is safe. The lifetime of the Adapter instance matches the scope of WriteableAuditLog::append() function. Actually that function accepts the same const reference. This could be unsafe only if Adapter will be used somewhere outside of WriteableAuditLog::append(). But that will be another story so I prefer to leave this as is.